### DISCUSSION TITLE
Is it possible to export information about the version of a port, used while building?

### DISCUSSION THREAD

**Discussion by Tagman** [microsoft/vcpkg#28990] [Q&A]
The goal is to be able to reproduce a specific build, down to exact versions of ports.

Ideally, I would like to do this, without pinning every single dependency in the vcpkg.json file. Instead always using the latest available ports.

I am imaging something like calling `npm list`

**Comment by Osyotr** [comment]
Use baselines https://learn.microsoft.com/en-us/vcpkg/reference/vcpkg-json#builtin-baseline

**Reply by Tagman** [reply]
This should work. But only in the case when I dont use overrides in my vcpkg.json file. I would be happier with a solution that explicitly lists the used versions.

Is there any other way to get a reliable list of used port versions?

**Reply by Osyotr** [reply]
> But only in the case when I dont use overrides in my vcpkg.json file

Why do you think so?

**Reply by Tagman** [reply]
If I understood it correctly, the baseline defines a commit in the vcpkg registry.
With this commit, it's basically pinning the possible port version I can use. From the first version to the version that is the newest, as seen by this commit.
Lets say a port at baseline a1b2c3 has possible versions: 1, 2, 3. If I am not specifying a version in my vcpkg.json it will always take version 3. Which I then can reproduce, by using the baseline.
But if I explicitly say I want to use version 2, via an override, I cannot say, after the build is done, which version was used.

I mean I could probably figure it out manually, by checking the vcpkg.json that was used for the build and then going through the dependencies and differentiating between "version-less" - using latest, and versioned dependencies - using the specified version.
But this seems like a PITA, only to be able to have a list of used versions.
